### PR TITLE
[testing] add ErrorInputs for `adaptive_{avg, max}_poolnd`

### PR DESCRIPTION
**Pull request by khushi-411**
Ref: https://github.com/pytorch/pytorch/pull/88906#discussion_r1040157313

Covers:
- [x] adaptive_avg_pool1d
- [x] adaptive_avg_pool2d
- [x] adaptive_avg_pool3d  
- [x] adaptive_max_pool1d
- [x] adaptive_max_pool2d
- [x] adaptive_max_pool3d

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by khushi-411** [comment]
Whoops, sorry about the cc's earlier and labels by github-actions. By mistake, I rebased with the incorrect branch. 😅😞

**Comment by khushi-411** [comment]
cc @mruberry @kshitij12345

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -2807,6 +2807,25 @@ def sample_inputs_adaptive_avg_pool1d(op_info, device, dtype, requires_grad, **k
...
         yield SampleInput(make_arg(input_shape[1:]), args=(output_size,))
 
+
+def error_inputs_adaptive_avg_pool1d(opinfo, device, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=torch.float32)
+
+    # error inputs for size overflow
+    # 0x0x3fffffffffffffff * 2 * 2 = 0xfffffffffffffffc = -4 as int64_t
+    # Tensor::numel() return int64_t, so following check that negative allocs are correctly handled
+    yield ErrorInput(SampleInput(make_arg((2, 2, 2)), kwargs={'output_size': 0x3fffffffffffffff}),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2817]: fyi you can now create SampleInput's "naturally" thanks to @peterbell10, so this could create the SampleInput like:

```
SampleInput(make_arg((2, 2, 2)), output_size=0x3fffffffffffffff)
```

File: test/nn/test_pooling.py [hunk 2]
```diff
@@ -1136,10 +1117,6 @@ def test_AvgPool3d_backward_after_cat_dim1_device(self, device):
 
     def test_pooling_size_empty(self, device):
         t = torch.rand([1, 2, 3, 4], device=device)
-        self.assertRaises(RuntimeError, lambda: F.adaptive_avg_pool1d(t, []))
-        self.assertRaises(RuntimeError, lambda: F.adaptive_avg_pool2d(t, []))
-        self.assertRaises(RuntimeError, lambda: F.adaptive_avg_pool3d(t, []))
-        self.assertRaises(RuntimeError, lambda: F.adaptive_max_pool1d(t, []))
         self.assertRaises(RuntimeError, lambda: F.adaptive_max_pool2d(t, []))
```
Comment (mruberry) [inline test/nn/test_pooling.py:1120]: Why are these error-cases left in this test?

File: test/nn/test_pooling.py [hunk 3]
```diff
@@ -158,14 +147,6 @@ def test_adaptive_pooling_size_none(self):
                 output = module(input)
                 self.assertEqual(output.size(), (4,) + (2,) * (numel - 1) + (4,))
 
-    @unittest.skipIf(TEST_WITH_UBSAN, "signed integer overflow error with UBSAN")
```
Comment (mruberry) [inline test/nn/test_pooling.py:161]: Should we skip the error inputs test when people are testing with a sanitizer?

Comment (khushi-411) [reply test/nn/test_pooling.py:161]: Thanks for the question, @mruberry. But just wondering why you think we need to ship this case?

Both raise errors for different reasons. Maybe adding a comment makes more sense. WDYT?
- **With UBSAN**: `integer multiplication overflow`
- **Without UBSAN**: `Storage size calculation overflowed`

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:2817]: Interesting! Will update the code. Thanks!

File: test/nn/test_pooling.py (hunk 2 above)
Comment (khushi-411) [reply test/nn/test_pooling.py]: Oops, added these cases too. My bad missed it earlier.

**Comment by khushi-411** [comment]
Hi, @mruberry! Gentle ping. I've updated the PR.

**Comment by khushi-411** [comment]
Hi, @mruberry! Gentle ping.

File: test/nn/test_pooling.py (hunk 3 above)
Comment (mruberry) [reply test/nn/test_pooling.py:161]: When PyTorch tests with its undefined behavior sanitizer, will `assertRaisesRegex` catch errors arising from the sanitizer?

However, I still think we should skip `test_errors` when running with a sanitizer. The test should be testing invalid inputs, after all. It seems like a lot of additional complexity to deal with a sanitizer when we're already on an invalid path that throws an error.

Comment (khushi-411) [reply test/nn/test_pooling.py]: @mruberry, I've updated the PR. Please take another look. Thanks!

**Review by mruberry — APPROVED** [review]
Cool! Thanks @khushi-411

**Comment by khushi-411** [comment]
Thanks for your insightful comments, @mruberry!

Merging this now :)

@pytorchmergebot merge
